app/models/asset.rb

```diff
@@ -162,7 +162,8 @@ def file_stat
   end
 
   def check_specified_replacement_exists
-    if replacement_id.present? && replacement.blank?
+    unscoped_replacement = Asset.unscoped.where(id: replacement_id)
```

Comment:
Minor: Using `Asset.unscoped.where` vs `Asset.unscoped.find_by` feels a bit odd here. I think we're only ever expecting at most a single asset to be returned so I think it would be better to use the latter.

Reply:
We can't easily use `Asset.unscoped.find_by` as that raises an exception when the record doesn't exist.